code/core/system/proxy-velodyne64/testsuites/ProxyVelodyne64TestSuite.h

```diff
@@ -22,21 +22,202 @@
 
 #include "cxxtest/TestSuite.h"
 
-// Include local header files.
-#include "../include/ProxyVelodyne64.h"
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include "opendavinci/generated/odcore/data/SharedPointCloud.h"
+#include "opendavinci/generated/odcore/data/pcap/GlobalHeader.h"
+#include "opendavinci/generated/odcore/data/pcap/Packet.h"
+#include "opendavinci/generated/odcore/data/pcap/PacketHeader.h"
+#include "opendavinci/odcore/data/Container.h"
+#include "opendavinci/odcore/io/conference/ContainerConference.h"
+#include "opendavinci/odcore/io/conference/ContainerListener.h"
+#include "opendavinci/odcore/io/protocol/PCAPProtocol.h"
+#include "opendavinci/odcore/wrapper/CompressionFactory.h"
+#include "opendavinci/odcore/wrapper/DecompressedData.h"
+#include "opendavinci/odcore/wrapper/SharedMemory.h"
+#include "opendavinci/odcore/wrapper/SharedMemoryFactory.h"
+
+#include "../include/velodyne64Decoder.h"
 
 using namespace std;
-using namespace opendlv::core::system::proxy;
+using namespace odcore::base;
+using namespace odcore::data;
+using namespace odcore::io::protocol;
+using namespace odcore::wrapper;
+
+class MyContainerConference : public odcore::io::conference::ContainerConference {
+   public:
+    MyContainerConference(std::shared_ptr< odcore::wrapper::SharedMemory > m)
+        : ContainerConference()
+        , m_counter(0)
+        , m_velodyneMemory(m) {}
+
+    virtual void send(odcore::data::Container &c) const {
+        if (c.getDataType() == odcore::data::SharedPointCloud::ID()) {
+        }
+        m_counter++;
+        if (m_counter == 2) { //We are only interested in the x, y, z, intensity values of points in the second Velodyne frame (Frame 1)
+            if (c.getDataType() == odcore::data::SharedPointCloud::ID()) {
+                odcore::data::SharedPointCloud velodyneFrame = c.getData< SharedPointCloud >();                                            //Get shared point cloud
+                std::shared_ptr< odcore::wrapper::SharedMemory > vsm = SharedMemoryFactory::attachToSharedMemory(velodyneFrame.getName()); //Attach the shared point cloud to the shared memory
```

Comment:
In this test suite, the shared memory segment is only attached once, i.e., when m_counter == 2. So it should be OK without such a flag. This flag is however used in the code for EnvironmentViewer in odcockpit